packages/core/src/balance/Balance.utils.ts

```diff
@@ -33,16 +58,99 @@ export function convertToTxUnit(balance: BN, power: number): BN {
+ * or it given a string checks for valid number representation.
+ *
+ * @param input [[BalanceNumber]] to convert.
+ * @returns String representation of the given [[BalanceNumber]].
+ * @throws On invalid number representation if given a string.
+ * @throws On malformed input.
+ */
+export function balanceNumberToString(input: BalanceNumber): string {
+  if (typeof input === 'string') {
+    if (!input.match(/^-?\d*\.?\d+$/)) {
+      throw new Error('not a string representation of number')
+    }
+    return input
+  }
+  if (
+    typeof input === 'number' ||
+    (typeof input === 'object' &&
+      ((input instanceof BN && input.toString) ||
+        (input instanceof BigInt && input.toLocaleString)))
+  ) {
+    return input.toString()
+  }
+  throw new Error('could not convert to String')
 }
+/**
+ * Converts the given [[BalanceNumber]] to the femto KILT equivalent.
+ *
+ * @param input [[BalanceNumber]] to convert.
+ * @param unit Metric prefix of the given [[BalanceNumber]].
+ * @returns Exact BN representation in femtoKilt, to use in transactions and calculations.
+ * @throws Unknown metricPrefix, or if the input has too many decimal places for it's unit.
+ */
+export function toFemtoKilt(
+  input: BalanceNumber,
+  unit: MetricPrefix = 'KILT'
+): BN {
+  const stringRepresentation = balanceNumberToString(input)
 
-export const TRANSACTION_FEE = convertToTxUnit(new BN(125), -9)
+  if (!Prefixes.has(unit)) {
+    throw new Error('Unknown metric prefix')
+  }
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const unitVal = Prefixes.get(unit)!
+  const negative = stringRepresentation.substring(0, 1) === '-'
+
+  const [integer, fraction] = negative
+    ? stringRepresentation.substring(1).split('.')
+    : stringRepresentation.split('.')
+  if (fraction && fraction.length > unitVal + 15) {
+    throw new Error(
+      `Too many decimal places: input with unit ${unit} and value ${stringRepresentation} exceeds the ${
+        unitVal + 15
+      } possible decimal places by ${fraction.length - unitVal + 15}`
+    )
+  }
+  const fractionBN = fraction
+    ? convertToTxUnit(new BN(fraction), unitVal - fraction.length)
+    : new BN(0)
+  const resultingBN = convertToTxUnit(new BN(integer), unitVal).add(fractionBN)
+
+  return resultingBN.mul(new BN(negative ? -1 : 1))
+}
+/**
+ * Converts the given [[BalanceNumber]] to a human readable number with metric prefix and Unit.
+ * This function uses the polkadot formatBalance function,
+ * it's output can therefore be formatted via the polkadot formatting options.
+ *
+ * @param input [[BalanceNumber]] to convert from Femto Kilt.
+ * @param options [[BalanceOptions]] for internationalization and formatting.
+ * @returns String representation of the given [[BalanceNumber]] with unit und metric prefix.
+ */
+export function fromFemtoKilt(
+  input: BalanceNumber,
+  options?: BalanceOptions
```

Comment:
Please add the an option to set the minimum fraction digits

Reply:
sadly this does not work as polkadotjs formatBalance always returns only up to 6 relevant digits, and they ceil so maximum of 3 digits are before the decimal point.

https://github.com/polkadot-js/common/blob/77289d0ec02bcb4714099593a8c3712df5f0606b/packages/util/src/format/formatBalance.ts#L68